causalml/metrics/sensitivity.py

```diff
@@ -0,0 +1,475 @@
+import logging
+import numpy as np
+import pandas as pd
+from collections import defaultdict
+import matplotlib.pyplot as plt
+from importlib import import_module
+from causalml.inference.meta import BaseTLearner
+
+logger = logging.getLogger('sensitivity')
+
+
+class ConfoundingFunction(object):
```

Comment:
We don't need the `ConfoundingFunction` class. Let's make 4 confounding functions available directly and define a catalog dictionary with the `str` name as key and the function as value so that `Sensitivity.sensitivity_analysis()` and `SensitivitySelectionBias.__init__()` take `str` for the `confound` input as follows:
```python
confounding_functions = {'one_sided': one_sided, ...}
...
    def sensitivity_analysis(self, methods, sample_size=None,
                                            confound='one_sided', alpha_range=None):
        ...
        try:
            confound_func = confounding_functions[confound]
        except KeyError:
            raise NotImplementedError(f'Confounding function, {confound} is not implemented. Use one of {confounding_functions.keys()}')
```